### PR TITLE
[DE-348] Added "Preview Mode" toggler for the new catalog service

### PR DESCRIPTION
**Pull request by lu-c**
Added ability to toggle between the new catalog service and the old data page.

- The mode you choose will be stored in localstorage, and when you click "browse data", it will pick the mode you last viewed.
- Added hero banner under the covid one
- The link for feedback does not exist right now, so we will have to patch that through again later

Screen Shot 2022-05-11 at 2 25 13 PM
Screen Shot 2022-05-11 at 2 25 19 PM
Screen Shot 2022-05-11 at 2 25 26 PM

### CODE REVIEW

File: integration-tests/tests/preview-dataset.js
Comment (lu-c) [inline integration-tests/tests/preview-dataset.js:10]: Had to make this change because the new browser page is intended to replace the datasets page and the mockups have now removed the "browse and explore" tab

File: src/pages/library/dataBrowser-utils.js
Comment (nawatts) [inline src/pages/library/dataBrowser-utils.js]: I enabled the data browser using `configOverridesStore.set({ isDataBrowserVisible: true })` and reloaded. The UI crashed with:
```
Uncaught TypeError: Cannot read properties of null (reading 'undefined')
    at getDatasetsPath (dataBrowser-utils.js:107:1)
    at LandingPage (LandingPage.js:67:1)
    ...
```

It worked after running `localStorage.setItem('catalog-beta-preview', '{}')`.

Comment (nawatts) [reply src/pages/library/dataBrowser-utils.js]: The Notebooks/Analyses tab has a similar "preview a new feature" button. In that case, the state is stored in config overrides / session storage. Is there a preference for one way or the other? Using config overrides instead of this new store would also avoid the complexity of passing the user object around / storing the user ID in local storage.

https://github.com/DataBiosphere/terra-ui/blob/f4b5658dbeaa275a8599fa855e5d696e5a163ed6/src/pages/workspaces/workspace/Notebooks.js#L454-L465

File: src/components/TopBar.js
Comment (nawatts) [inline src/components/TopBar.js:218]:
```suggestion
            }, ['Data Catalog (BETA)']),
```

A phrase here seems out of place since the other options are all nouns.

File: src/pages/library/DataBrowserToggler.js
Comment (nawatts) [inline src/pages/library/DataBrowserToggler.js:15]: Should this component be rendered at all if the data browser is not enabled?

Otherwise, could this be simplified to avoid rendering the empty div?

```suggestion
  return !!isDataBrowserVisible() && div({
```

File: src/pages/library/DataBrowserToggler.js
Comment (nawatts) [inline src/pages/library/DataBrowserToggler.js:30]: Is a Switch the best interface to use for this? I think it would be surprising for changing a checkbox to cause page navigation. Would a link or button be more appropriate?

File: src/pages/library/DataBrowserToggler.js
Comment (nawatts) [inline src/pages/library/DataBrowserToggler.js]: This says "On" regardless of whether or not the beta catalog is enabled or not.

File: src/pages/LandingPage.js
Comment (nawatts) [inline src/pages/LandingPage.js:123]: Is there a way to reword this so that the phrase "Preview the BETA Data Catalog" isn't repeated?

To be more consistent with the COVID banner above it, maybe something like "BETA Data Catalog" as the heading and below that in normal text, "Preview the Data Catalog and provide valuable feedback." with "Preview the Data Catalog" or the whole sentence linked to the catalog?

**Review by nawatts — CHANGES_REQUESTED** [review]
A few things, but mainly requesting changes to handle the case where `catalog-beta-preview` is not set in local storage. Currently, this causes the UI to crash. To reproduce, run `localStorage.removeItem('catalog-beta-preview')`.

File: src/pages/library/DataBrowserToggler.js
Comment (lu-c) [reply src/pages/library/DataBrowserToggler.js:15]: I tried doing that initially, and it complained that this function must return an object!

File: src/components/TopBar.js
Comment (lu-c) [reply src/components/TopBar.js:218]: I want to stick with the mock ups UX gave us for this wording, and revisit it later. Right now, at least, this feature is not ready to be turned on anyway, so there will be things we need to do before that too

File: src/pages/LandingPage.js
Comment (lu-c) [reply src/pages/LandingPage.js:123]: I like this, going to make that change! :)

File: src/pages/library/DataBrowserToggler.js
Comment (lu-c) [reply src/pages/library/DataBrowserToggler.js:30]: That's true, I think it does feel kind of unexpected... but in the browser, it's not entirely obvious that it's a new page loaded, does it? It looks like you are just toggling a setting, since the tab doesnt actively look like it's changed

File: src/pages/library/DataBrowserToggler.js
Comment (lu-c) [reply src/pages/library/DataBrowserToggler.js]: Fixed

File: src/pages/library/dataBrowser-utils.js
Comment (lu-c) [reply src/pages/library/dataBrowser-utils.js]: Fixed this bug. I'd be okay with making that change to use a new configOverridesStore, just trying to think through what the pros and cons of this would be. I'm using the localstorage because I think that's also tied to user... Do we care if someone logs out, the next person who uses the same browser/computer ends up getting the same state? Since you can toggle it back easily, I dont think it's that important 🤔 

The other thing that I'm also thinking about is that this is only supposed to be around as long as we are maintaining both versions. The catalog is supposed to eventually replace it, which means this localstorage vs configoverrides is eventually all going to have to go away.

Dunno! Looking for other opinions on this too! Thanks :)

File: src/pages/library/DataBrowserToggler.js
Comment (nawatts) [reply src/pages/library/DataBrowserToggler.js:30]: It doesn't _look_ like it, but the navigation is noticeable in other ways. Focus gets moved back to the top of the page and if you're using a screen reader, it reads the new page. I leave it up to you which way to go.

However, one thing I noticed while trying this with a screen reader is that the switch doesn't have a label. It currently reads as "off, switch" or "on, switch". There's already some descriptive text here, it just needs to be wrapped in a `label` tag with the Switch component. The usage section on https://www.npmjs.com/package/react-switch has an example.

File: src/pages/LandingPage.js
Comment (nawatts) [inline src/pages/LandingPage.js]: Ending this sentence with a period would make it consistent with the COVID section.

**Review by nawatts — CHANGES_REQUESTED** [review]
One last thing... the toggle switch needs an accessible label.

**Comment by s-rubenstein** [comment]
I think to link this to Jira it should be `[DC-348]` rather than `[DE-348]` in the title

[DC-348]: https://broadworkbench.atlassian.net/browse/DC-348?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: src/pages/LandingPage.js
Comment (s-rubenstein) [inline src/pages/LandingPage.js]: Do you think it would be worth extracting the card width into a variable, and then making this more explicit/removing the comment and making it:
`width: 'calc(cardWidth * 3 + 'px + ' cardMargin * 2 + 'rem')'`?

That would have the benefit of automatically changing this to match if the cards are changed.

File: src/pages/library/DataBrowserToggler.js
Comment (fboulnois) [inline src/pages/library/DataBrowserToggler.js]:
```suggestion
        href: '#', // TODO: Add when this is added.
```

Should this be something other than empty?

**Review by fboulnois — COMMENTED** [review]
I like Sky's suggestion. One other question:

File: src/pages/library/DataBrowserToggler.js
Comment (lu-c) [reply src/pages/library/DataBrowserToggler.js]: O, good call!

File: src/pages/LandingPage.js
Comment (lu-c) [reply src/pages/LandingPage.js]: Updated!

File: src/pages/library/DataBrowserToggler.js
Comment (lu-c) [reply src/pages/library/DataBrowserToggler.js:30]: Ok, updated the label so it is now wrapping the content - Let me know if this is okay!

**Review by nawatts — APPROVED** [review]

**Review by fboulnois — APPROVED** [review]

**Comment by fboulnois** [comment]
I think it needs to be DC-348 to link to our Jira board, but otherwise lgtm.

File: src/pages/LandingPage.js
Comment (s-rubenstein) [inline src/pages/LandingPage.js:95]: I think you need to add `rem` to the end of the margin in the calc string?

File: src/pages/LandingPage.js
Comment (s-rubenstein) [inline src/pages/LandingPage.js:112]: Add rem to end of 2 in the calc string.

**Review by s-rubenstein — APPROVED** [review]
LGTM. One change, but the merge doesn't need to block on another round of reviews from me.

File: src/pages/LandingPage.js
Comment (nawatts) [reply src/pages/LandingPage.js:95]: This looks correct as is. `styles.card.marginRight` is a string with `rem` units. Thus, the other operand for multiplication must be a number (without units) vs a length.

https://developer.mozilla.org/en-US/docs/Web/CSS/calc#syntax

Comment (lu-c) [reply src/pages/LandingPage.js:95]: If you run this locally tho, it works - I wouldn't be pushing up code without testing it!

**Pull request merged**
